R/write_captions.R

```diff
@@ -269,17 +610,274 @@ write_captions <- function(dat, # converted model output object
+   'recruitment_start_year' = recruitment_start_year,
+   # 'recruitment_end_year' = recruitment_end_year,
+   'recruitment_min' = recruitment_min,
+   'recruitment_max' = recruitment_max,
+
+   ## recruitment deviations
+   'recruit_dev_start_year' = recruit_dev_start_year,
+   # 'recruit_dev_end_year' = recruit_dev_end_year,
+   'recruit_dev_min' = recruit_dev_min,
+   'recruit_dev_max' = recruit_dev_max,
+
+   ## tot_b (total biomass)
+   # 'biomass_start_year' = biomass_start_year,
+   # 'biomass_end_year' = biomass_end_year,
+   # 'biomass_units' = biomass_units,
+   # 'biomass_min' = biomass_min,
+   # 'biomass_max' = biomass_max,
+   # 'biomass_ref_pt' = biomass_ref_pt,
+   # 'biomass_ref_pt_units' = biomass_ref_pt_units,
+
+   # relative ssb
+   # NOTE: moving this above recruitment so rel_ssb_min isn't changed to
+   # "rel_" + ssb_min, etc.
+   'rel_ssb_min' = rel_ssb_min,
+   'rel_ssb_max' = rel_ssb_max,
+
+   ## spawning_biomass (ssb)
+   'ssb_start_year' = ssb_start_year,
+   # 'ssb_end_year' = ssb_end_year,
+   # 'ssb_units' = ssb_units,
+   'ssb_min' = ssb_min,
+   'ssb_max' = ssb_max#,
+   # 'ssb_ref_pt' = ssb_ref_pt,
+   # 'ssb_ref_pt_units' = ssb_ref_pt_units,
+   # 'ssbtarg' = ssbtarg,
+
+
+   # ## spr (spawning potential ratio)
+   # 'spr_start_year' = spr_start_year,
+   # 'spr_end_year' = spr_end_year,
+   # 'spr_min' = spr_min,
+   # 'spr_max' = spr_max,
+   # 'spr_ref_pt' = spr_ref_pt,
+   # 'spr_ref_pt_units' = spr_ref_pt_units,
+   #
+   # ## pop_naa_baa (population numbers at age and population biomass at age)
+   # 'pop_naa_baa_start_year' = pop_naa_baa_start_year,
+   # 'pop_naa_baa_end_year' = pop_naa_baa_end_year,
+   # 'pop_naa_baa_fish_min' = pop_naa_baa_fish_min,
+   # 'pop_naa_baa_fish_max' = pop_naa_baa_fish_max,
+   #
+   # ## proj_catch (projected catch)
+   # 'proj_catch_units' = proj_catch_units,
+   # 'proj_catch_start_year' = proj_catch_start_year,
+   # 'proj_catch_end_year' = proj_catch_end_year,
+   # 'proj_catch_min' = proj_catch_min,
+   # 'proj_catch_max' = proj_catch_max,
+   #
+   # ## proj_biomass (projected biomass)
+   # 'proj_biomass_units' = proj_biomass_units,
+   # 'proj_biomass_start_year' = proj_biomass_start_year,
+   # 'proj_biomass_end_year' = proj_biomass_end_year,
+   # 'proj_biomass_min' = proj_biomass_min,
+   # 'proj_biomass_max' = proj_biomass_max,
+   # 'proj_biomass_ref_pt' = proj_biomass_ref_pt,
+   # 'proj_biomass_ref_pt_units' = proj_biomass_ref_pt_units,
+   #
+   # ## Other
+   # 'tot_catch' = tot_catch,
+   # 'M' = M,
+   # 'steep' = steep,
+   # 'SBmsy' = SBmsy,
+   # 'fSB' = fSB,
+   # 'sbtarg' = sbtarg,
+   #
+   # # TABLES-----
+   #
+   # ## catch
+   # 'catch_fleet' = catch_fleet,
+   #
+   # ## landings
+   # 'landings_tbl_units' = landings_tbl_units,
+   #
+   # ## discards
+   # 'discards_tbl_units' = discards_tbl_units,
+   #
+   # ## catchability
+   # 'catchability_fleet' = catchability_fleet
+  )
+
+  caps_alttext_subbed <- caps_alttext |>
```

Comment:
Can you add a note explaining what this pipeline is doing? I also haven't see the ", ~{" notation before in the below line

Reply:
Sure, added the note. The internet helped me with chunk, so here's what I believe is going on: the ~ is present because an anonymous function is passed to another function (dplyr::mutate). The { is separating the for loop from the output of the for loop, which is acted upon by dplyr::mutate.